modules/repository/repo.go

```diff
@@ -300,3 +313,67 @@ func PushUpdateAddTag(repo *models.Repository, gitRepo *git.Repository, tagName
 
 	return models.SaveOrUpdateTag(repo, &rel)
 }
+
+// StoreMissingLfsObjectsInRepository downloads missing LFS objects
+func StoreMissingLfsObjectsInRepository(ctx context.Context, repo *models.Repository, gitRepo *git.Repository, endpoint *url.URL) error {
+	client := lfs.NewClient(endpoint)
+	contentStore := lfs.NewContentStore()
+
+	pointerBlobs, err := lfs.SearchPointerBlobs(gitRepo)
+	if err != nil {
+		return err
+	}
+
+	for _, pointerBlob := range pointerBlobs {
```

Comment:
pointerBlobs may be quite long really should be reading each blob off a stream or even channel instead.

Reply:
A channel would be useful only if it supports cancellation? The original code was moved from [here](https://github.com/go-gitea/gitea/blob/cc6470425c720a05abc55b21f65e3d8179618900/routers/repo/lfs.go#L411) ([list creation](https://github.com/go-gitea/gitea/blob/cc6470425c720a05abc55b21f65e3d8179618900/routers/repo/lfs.go#L438)). Should this change request be part of this PR?